### PR TITLE
Enhancements for TaskProActiveDataspaces

### PR DESCRIPTION
**Pull request by lpellegr**

### CODE REVIEW

**Comment by lpellegr** [comment]
@ow2-proactive/developers Could two people please review the changes?

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 1]
```diff
@@ -94,13 +94,20 @@
             Executors.newFixedThreadPool(getFileTransferThreadPoolSize(),
                     new NamedThreadFactory("FileTransferThreadPool"));
 
+    /**
+     * Mainly for testing purposes
```
Comment (MaelAudren) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: If the aim of the visibility is to test then the visibility "package" is more accurate than the visibility "protected".

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 2]
```diff
@@ -458,10 +455,39 @@ private void createFolderHierarchySequentially(
...
+            DataSpacesFileObject parent = target.getParent();
+
+            if (isDebugEnabled) {
+                logger.debug("Creating folder " + parent.getRealURI());
+            }
+
+            parent.createFolder();
+        }
+    }
+
```
Comment (MaelAudren) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: Why if the property value is null then you return true ?

**Comment by MaelAudren** [comment]
review done

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 3]
```diff
@@ -94,13 +94,20 @@
             Executors.newFixedThreadPool(getFileTransferThreadPoolSize(),
                     new NamedThreadFactory("FileTransferThreadPool"));
 
+    /**
+     * Mainly for testing purposes
+     */
+    protected TaskProActiveDataspaces() {
+
+    }
+
```
Comment (MaelAudren) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: If the aim of the visibility is to test then the visibility "package" is more accurate than the visibility "protected".

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java [hunk 4]
```diff
@@ -458,10 +455,39 @@ private void createFolderHierarchySequentially(
...
+            parent.createFolder();
+        }
+    }
+
+    private boolean isCreateFolderHierarchySequentiallyEnabled() {
+        String property = System.getProperty(PA_NODE_DATASPACE_CREATE_FOLDER_HIERARCHY_SEQUENTIALLY);
+
+        return property == null || "true".equalsIgnoreCase(property);
+    }
+
```
Comment (MaelAudren) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: Why if the property value is null then you return true ?

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java (hunk 3 above)
Comment (lpellegr) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: Suggestion applied.

File: scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java (hunk 4 above)
Comment (lpellegr) [inline scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/data/TaskProActiveDataspaces.java]: By default the property is not defined. In such a case I want the create folder hierarchy sequentially feature to be enabled. The property is there to have a mean to disable this last  feature.

**Comment by lpellegr** [comment]
@MaelAudren Thank you for the review. 

@ow2-proactive/developers One more review please.

**Comment by tobwiens** [comment]
:+1: Looks good. 
I have one remark: I see one test with @Test(expected = Exception.class). Might it be important to test specific exceptions under specific conditions?

**Pull request merged**